resource/authentication_manager.go

```diff
@@ -37,66 +37,79 @@ func (a *AuthenticationManager) GetCredentialPath(key string) string {
 
 // Create a new container authentication credential and write it into the Agent's host file system. The credential file
 // live in a directory named by the key input parameter.
-func (a *AuthenticationManager) CreateCredential(key string, id string, ver string) error {
+func (a *AuthenticationManager) CreateCredential(key string, id string, ver string, isDev bool) error {
```

Comment:
The variable isDev, I would prefer that it was named, secureCreds or something like that which communicates that the caller is asking the function to create secure creds or not. The fact that hzn dev uses insecure creds is just one example (happens to be the only example today) where this feature is needed. This comment and the next one also apply to the RemoveCredential function.